server/sonar-server/src/main/java/org/sonar/server/es/IndexTypeId.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.es;
+
+import java.util.Arrays;
+import java.util.function.Function;
+import org.sonar.core.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class IndexTypeId {
+
+  private final String index;
+  private final String type;
+
+  public IndexTypeId(String index, String type) {
+    this.index = requireNonNull(index);
+    this.type = requireNonNull(type);
+  }
+
+  public String getIndex() {
+    return index;
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public static String[] getIndices(IndexTypeId... indexTypes) {
+    return getDetails(IndexTypeId::getIndex, indexTypes);
+  }
+
+  public static String[] getTypes(IndexTypeId... indexTypes) {
+    return getDetails(IndexTypeId::getType, indexTypes);
+  }
+
+  private static String[] getDetails(Function<? super IndexTypeId, ? extends String> function, IndexTypeId... indexTypes) {
+    return Arrays.stream(indexTypes).map(function).collect(Collectors.toSet(indexTypes.length)).toArray(new String[0]);
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((index == null) ? 0 : index.hashCode());
+    result = prime * result + ((type == null) ? 0 : type.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 8, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1142)